### ISSUE TITLE
Enhance python setup for on-prem

### ISSUE THREAD

**Issue by ZollnaPa** [IBMStreams/streamsx.topology#1611]
The recommended anaconda installation provides overhead of tools. The python installation is thinner than anaconda. The installation instruction of the python bundle does not provide all detail that are required for the usage of the topology toolkit. To run the topology tester, some additional tasks are required. The python version has to be adequate to anaconda installation. it means the python  3.5 version is required.

-  the installation commands must use --enabled-shared option and should use the --prefix option for alternative installation, e.g.:

      ./configure --prefix=/usr/local --enable-shared
      make
      make altinstall

- the PYTHONHOME should be set now to the installation path, e.g..

      /usr/local/Python3.5.2

- usual, the command call for python is defined as 'python3'. It required a link creation in the bin directory of the PYTHONHOME path, e.g.:

      ln -s ./python3.5 /usr/local/Python3.5.2/bin/python3

- the same for pip and python3-config:

      ln -s ./pip3.5 /usr/local/Python3.5.2/bin/pip
      ln -s ./python3.5m-config /usr/local/Python3.5.2/bin/python3-config

- the python libraries must be set to library path. The best way is the daemon setting as root, e.g.:

      echo /usr/local/Python3.5.2/lib >/etc/ld.so.conf.d/rh-python3.5.conf
      ldconfig

- the topology tester requires the update for following modules:

      pip install ibm-cos-sdk
      pip install future

The required setting of the application environment variable PYTHONHOME as a streams property is already described.

**Comment by ddebrunner** [comment]
Why does the topology tester rely on `ibm-cos-sdk`?

**Comment by ddebrunner** [comment]
I'm not sure what this issue is suggesting?

For example if you create a Python environment that uses the command `python3.5` then there's no requirement to have a `python3`. The `streamsx` package and its tester don't impose such a requirement do they?

Also the tester is supported on Python 2.7, so there's no requirement to use 3.5.

**Comment by ddebrunner** [comment]
Can you explain more the update to `future`?  The `streamsx` package has a dependency on `future`, does it need specify a minimum version?

**Comment by ZollnaPa** [comment]
OK, the ibm-cos-sdk is the requirement,that is relevant for object storage. You are right, the topology tester needs the module to test COS. This should be moved to toolkit.objectstorage test description if not specified now.

Regarding the python3 linking, The anaconda provides the link to python3.5. I mean, it is useful to get the same execution call for anaconda as well as for python 3.5 installation.

The last is the future module. While I started the test, the 'past' object was required. I found, that the future module needs to be installed for the 'past' object.

My general opinion is, you should continue with the documentation of the setup configuration as you started with PYTHONHOME and --enable-shared option. The setup 'hints' that I found were not sufficient to use the topology tester in a local installed python 3.5 environment. The steps, that I described are not direct requirements to topology toolkit but they help to use topology in the python 3.5 environment.

**Comment by ZollnaPa** [comment]
I installed the future module just as I described: 'pip install future'. The current version is 0.16.0. Any version dependency was not pointed during the installation.
this is the list of modules in my python 3.5 environment:
Package                Version  
---------------------- ---------
certifi                2018.4.16
chardet                3.0.4    
docutils               0.14     
future                 0.16.0   
ibm-cos-sdk            2.1.0    
ibm-cos-sdk-core       2.1.0    
ibm-cos-sdk-s3transfer 2.1.0    
idna                   2.6      
jmespath               0.9.3    
pip                    10.0.0   
python-dateutil        2.7.2    
requests               2.18.4   
setuptools             20.10.1  
six                    1.11.0   
urllib3                1.22
